pointlesspun/StoryNode
# V03updates

Implemented:


* update read me
* Refactor (app-main, introduce app-io)
* Add some spacing between the buttons
* choose examples from front page
* Create a (no: small maze game) set of tests instead of the horror story (store story examples for now)
* move data from src
* Add random conditionals for some variation
* Add function evaluations to conditionals
* test else and alternatives
* Make sure the option buttons appear in the same order as in the md file
* Conditional links are really necessary - as it shows with the camera feed story parts, it will revisit parts that already have been disclosed. Current considerations are:
    * add variables to the code: `set(variableName, value)` and `compare(variableName, operator, value)`. Then have a convention where a link has the syntax 
        * [conditional expression -> name](reference)
    If the conditional expression is not true, the link will not show up. This setup makes it so, the links should still work in md.
* create a node structure for conditionals
* Add "file name" to save state - load the current state matching the name { "file": "state", "file2": "state2" } 
* integration conditional options into the story node (and update the tests)
* Test AST for option conditionals
* red path boolean (?) / conditionals
* Save state and continue where left off
* Move delay table to data as well as dictionary begin and end tokens
* add webpack
* clean-up: remove unused deprecated methods from text-plotter   
* bug: pause after a automatically delayed character (, or .) goes straight into pause and does not handle the delay
* bug: the way plot delay is handled in updatePlotText is wrong, setting the speed to something high has therefore no effect. 
* Remove time calls from plotter
* Remove callbacks in favor of a queue avoiding potential endless circular calls
* add tests for orchestration and main methods
* Download fonts and other resources in a different way than via google fonts href if possible.
